sconfig/shared/src/main/scala/org/ekrich/config/impl/AbstractConfigValue.scala

```diff
@@ -316,47 +316,56 @@ abstract class AbstractConfigValue private[impl] (val _origin: ConfigOrigin)
     getClass.getSimpleName + "(" + sb.toString + ")"
   }
 
-  private[impl] def render(
+  private[impl] def renderAtKey(
       sb: jl.StringBuilder,
       indent: Int,
       atRoot: Boolean,
-      atKey: String,
+      atKey: String, // nullable
       options: ConfigRenderOptions
   ): Unit = {
-    if (atKey != null) {
-      val renderedKey =
-        if (options.getJson) ConfigImplUtil.renderJsonString(atKey)
-        else ConfigImplUtil.renderStringUnquotedIfPossible(atKey)
-      sb.append(renderedKey)
-      if (options.getJson) {
-        if (options.getFormatted)
-          sb.append(" : ")
-        else
-          sb.append(":")
-      } else { // in non-JSON we can omit the colon or equals before an object
-        if (this.isInstanceOf[ConfigObject]) {
-          if (options.getFormatted) sb.append(' ')
-        } else {
-          sb.append(
-            if (options.formattingOptions.colonAssign) ":" else "="
-          )
-        }
+    Option(atKey)
+      .foreach { key =>
+        val renderedKey =
+          if (options.getJson) ConfigImplUtil.renderJsonString(key)
+          else ConfigImplUtil.renderStringUnquotedIfPossible(key)
+        renderAtRenderedKey(sb, renderedKey, options)
       }
-    }
-    render(sb, indent, atRoot, options)
+    renderValue(sb, indent, atRoot, options)
   }
-  def render(
+
+  def renderValue(
       sb: jl.StringBuilder,
       indent: Int,
       atRoot: Boolean,
       options: ConfigRenderOptions
-  ): Unit =
-    if (hideEnvVariableValue(options)) {
-      sb.append("<env variable>")
-    } else {
-      val u = unwrapped
-      sb.append(u.toString)
+  ): Unit = if (hideEnvVariableValue(options)) {
+    sb.append("<env variable>")
+  } else {
+    val u = unwrapped
+    sb.append(u.toString)
+  }
+
+  private[impl] def renderAtRenderedKey(
+      sb: jl.StringBuilder,
+      renderedKey: String, // not nullable
+      options: ConfigRenderOptions
+  ): Unit = {
+    sb.append(renderedKey)
+    if (options.getJson) {
+      if (options.getFormatted)
+        sb.append(" : ")
```

Comment:
I'm a little confused here. All the original was with no spacing and here we have space on both sides?

Reply:
good catch, lets don't add special entry to add spaces to HOCON style.
Just when formatted is true then add those spaces always.
Simpler and for JSON it was already happening.